### ISSUE TITLE
Jackson 2.1x / 2.6.x version clash

### ISSUE THREAD

**Issue by ronhitchens** [docker-java/docker-java#433]
Commit: 1f943a3df92eb87e24f0dd0ffdc2c5898ffb6f8d 

This commit has broken my builds. We're using Docker-Java to manage docker images for automated testing. Our app uses the RatPack framework, which depends on Jackson 2.6.x. Unfortunately, Jackson 2.1.x and 2.6.x are incompatible with each other in the same JVM. That means we couldn't use the released version of Docker-Java since it depends on Jackson 2.1.x. We've been using the Docker-Java 3.0.0-SNAPSHOT versions to get around this problem because it was using the current version of Jackson.

But with this dependency downgrade, we can no longer use Docker-Java with any code that requires a newer version of Jackson. I strongly use you to break this dependency on an older version of Jackson. 

Thanks.

=====================

```
javax.ws.rs.ProcessingException: com.fasterxml.jackson.databind.ObjectWriter.getJsonFactory()Lcom/fasterxml/jackson/core/JsonFactory;
	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:255)
	at org.glassfish.jersey.client.JerseyInvocation$2.call(JerseyInvocation.java:683)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:228)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:424)
	at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:679)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:435)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.post(JerseyInvocation.java:338)
	at com.github.dockerjava.jaxrs.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:33)
	at com.github.dockerjava.jaxrs.CreateContainerCmdExec.execute(CreateContainerCmdExec.java:15)
	at com.github.dockerjava.jaxrs.AbstrSyncDockerCmdExec.exec(AbstrSyncDockerCmdExec.java:23)
	at com.github.dockerjava.core.command.AbstrDockerCmd.exec(AbstrDockerCmd.java:35)
	at com.github.dockerjava.core.command.CreateContainerCmdImpl.exec(CreateContainerCmdImpl.java:123)
	at com.github.dockerjava.api.command.CreateContainerCmd$exec$0.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:110)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:114)
	at tests.lib.DockerMarkLogic.setup(DockerMarkLogic.groovy:99)
	at tests.lib.DockerMarkLogic$setup.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:110)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:114)
	at tests.DockerContainerResource.before(DockerContainerResource.groovy:61)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:46)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: java.lang.NoSuchMethodError: com.fasterxml.jackson.databind.ObjectWriter.getJsonFactory()Lcom/fasterxml/jackson/core/JsonFactory;
	at com.fasterxml.jackson.jaxrs.json.annotation.EndpointConfig.initWriter(EndpointConfig.java:159)
	at com.fasterxml.jackson.jaxrs.json.annotation.EndpointConfig.forWriting(EndpointConfig.java:63)
	at com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider.writeTo(JacksonJsonProvider.java:531)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.invokeWriteTo(WriterInterceptorExecutor.java:265)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor$TerminalWriterInterceptor.aroundWriteTo(WriterInterceptorExecutor.java:250)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162)
	at com.github.dockerjava.jaxrs.filter.LoggingFilter.aroundWriteTo(LoggingFilter.java:301)
	at org.glassfish.jersey.message.internal.WriterInterceptorExecutor.proceed(WriterInterceptorExecutor.java:162)
	at org.glassfish.jersey.message.internal.MessageBodyFactory.writeTo(MessageBodyFactory.java:1154)
	at org.glassfish.jersey.client.ClientRequest.writeEntity(ClientRequest.java:503)
	at com.github.dockerjava.jaxrs.connector.ApacheConnector$2.writeTo(ApacheConnector.java:576)
	at org.apache.http.impl.execchain.RequestEntityProxy.writeTo(RequestEntityProxy.java:116)
	at org.apache.http.impl.DefaultBHttpClientConnection.sendRequestEntity(DefaultBHttpClientConnection.java:155)
	at org.apache.http.impl.conn.CPoolProxy.sendRequestEntity(CPoolProxy.java:149)
	at org.apache.http.protocol.HttpRequestExecutor.doSendRequest(HttpRequestExecutor.java:236)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:121)
	at org.apache.http.impl.execchain.MainClientExec.execute(MainClientExec.java:254)
	at org.apache.http.impl.execchain.ProtocolExec.execute(ProtocolExec.java:195)
	at org.apache.http.impl.execchain.RetryExec.execute(RetryExec.java:86)
	at org.apache.http.impl.execchain.RedirectExec.execute(RedirectExec.java:108)
	at org.apache.http.impl.client.InternalHttpClient.doExecute(InternalHttpClient.java:184)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:72)
	at com.github.dockerjava.jaxrs.connector.ApacheConnector.apply(ApacheConnector.java:441)
	at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:246)
	... 35 more
```

**Comment by marcuslinke** [comment]
Sorry for this. It was an oversight of me. I have currently no time to fix it. If you could provide a simple PR I will merge urgently.

**Comment by ronhitchens** [comment]
I don't currently have time to fix it either.  I am using an explicit dependency on on a snapshot build from 2016-01-18.  When you've made the fix, I will go back to 3.0.0-SNAPSHOT to get the latest.

**Comment by KostyaSha** [comment]
@ronhitchens please remember that library can't satisfy dependencies for all projects and frameworks. Technically if it passing tests, then it not library issue and you can bypass it by using shaded library version.

**Comment by ronhitchens** [comment]
@KostyaSha Yes, I agree that the Docker-Java library isn't at fault.  But as a practical matter it is problematic if it depends on a relatively old version of Jackson.  When using Docker-java with other libraries that require a newer version of Jackson then there is a problem because Jackson is not API compatible between versions 2.1.x and 2.6.x of itself.  Not your fault but a problem still.

If there is a way to use both versions at once in a single JVM, I'd be glad to know how to do that.

Given that until a couple of days ago Docker-Java was working fine with the latest Jackson, it behooves you to continue on that path rather than causing incidental breakage but reverting to a dependency on an older version of Jackson.

**Comment by marcuslinke** [comment]
Fixed via 9b3ed93add255b7da3e123f5a8710fa54c9ca651
